accounts-db/src/accounts_index/bucket_map_holder.rs

```diff
@@ -82,6 +90,49 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> BucketMapHolder<T, U>
         self.disk.is_some()
     }
 
+    /// Check if flushing to disk should occur based on entry count threshold per bin
+    ///
+    /// # Arguments
+    /// * `entries_in_bin` - Current number of entries in this bin
+    /// * `high_water_mark_ratio` - Ratio (0.0-1.0) of per-bin threshold to trigger flushing
+    ///
+    /// Returns true if:
+    /// - IndexLimit::Minimal (always flush)
+    /// - IndexLimit::Threshold and entries_in_bin > high_water_mark_ratio * threshold_entries_per_bin
+    pub fn should_flush_to_disk(&self, entries_in_bin: usize, high_water_mark_ratio: f64) -> bool {
+        match self.threshold_entries_per_bin {
+            None => {
+                // Check if disk index is enabled - if so, always flush (Minimal mode)
+                self.is_disk_index_enabled()
+            }
+            Some(threshold_per_bin) => {
+                // Threshold mode: compare entries in this bin against per-bin threshold
+                let high_water_mark = (threshold_per_bin as f64 * high_water_mark_ratio) as usize;
+                entries_in_bin > high_water_mark
+            }
+        }
+    }
```

Comment:
IMO the threshold field should be the actual high watermark value. Then we don't need to do this math every time. (And I was surprised that the threshold field wasn't actually the threshold.)

Reply:
rename the member to be target_entries_per_bin. 
I think it is good to keep the high_water_mark_ratio here. It gives us more freedom. 
Maybe in future, we want to dynamically change high_water_mark_ratio.